src/main/java/uk/ac/ebi/intact/app/internal/ui/components/query/AdvancedSearchQueryComponent.java

```diff
@@ -0,0 +1,261 @@
+            fullQuery.append(" ")
+                     .append(queryOperators.getRuleOperator())
+                     .append(" ");
+        }
+
+        for (RuleSetBuilder ruleSetBuilder : ruleSetBuilders) {
+            fullQuery.append(ruleSetBuilder.getQuery());
+        }
+
+        return fullQuery.toString();
+    }
+
+    private JScrollPane getRuleScrollPane() {
+        JScrollPane scrollPane = new JScrollPane();
+
+        rulesPanel.setAutoscrolls(true);
+        rulesPanel.setLayout(new BoxLayout(rulesPanel, BoxLayout.Y_AXIS));
+
+        scrollPane.setVerticalScrollBarPolicy(JScrollPane.VERTICAL_SCROLLBAR_ALWAYS);
+        scrollPane.setHorizontalScrollBarPolicy(JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+        scrollPane.setViewportView(rulesPanel);
+
+        return scrollPane;
+    }
+
+    public void addRule(){
+        JPanel rulePanel = new JPanel();
+        OneRuleBuilderPanel oneRuleBuilderPanel = new OneRuleBuilderPanel(this);
+        rules.add(oneRuleBuilderPanel);
+
+        rulePanel.add(oneRuleBuilderPanel.getOneRuleBuilderPanel());
+        rulePanel.add(getDeleteRuleButton(rulesPanel));
+
+        queryTextField.setText(getQueriesFromRuleBuilders(rules, queryOperators.getRuleOperator()));
+        rulesPanel.add(rulePanel);
+        rulesPanel.revalidate();
+        rulesPanel.repaint();
+    }
+
+    private JButton getDeleteRuleButton(JPanel rulePanel) {
+        JButton deleteRuleButton = new JButton("Delete rule");
+        setButtonIntactPurple(deleteRuleButton);
+        deleteRuleButton.addActionListener(e -> {
+            deleteRule(rulePanel);
+        });
+        return deleteRuleButton;
+    }
+
+    public void deleteRule(JPanel oneRuleBuilder) {
+        rulesPanel.remove(oneRuleBuilder);
+
+        queryTextField.setText(getQueriesFromRuleBuilders(rules, queryOperators.getRuleOperator()));
+        rulesPanel.revalidate();
+        rulesPanel.repaint();
+    }
+
+    private JPanel getQueryInputField(){
+        JPanel queryInputFieldContainer = new JPanel();
+
+        queryTextField.setMinimumSize(new Dimension(frameWidth, 25));
+        queryTextField.setPreferredSize(new Dimension(frameWidth/2, 25));
+        queryTextField.setVisible(true);
+
+        queryInputFieldContainer.add(queryTextField);
+
+        return queryInputFieldContainer;
+    }
+
+    public String getQueryFromTextField(){
+        return queryTextField.getText();
+    }
+
+    private void modifyComboBoxesFromTextField(String queryFromTextField){
+        //todo: remove every combo boxes and build from scratch
+
+        String[] queries = getQueriesFromTextField(queryFromTextField);
+        modifyQueryBuildersNumberFromTextField(queries);
+
+        for (int i = 0; i < queries.length; i++) {
+            for (Field field : Field.FIELD_MAP.values()) {
+                if (queryFromTextField.contains(field.getMiqlQuery())) {
+                    QueryComponents queryComponents = parseQuery(queries[i].trim());
```

Comment:
We're not doing anything with `queryComponents` after it's assign. What's the purpose of these loops?